index.html

```diff
@@ -920,5 +920,69 @@ <h3>
               </dd>
             </dl>
 </section>
+<section>
+    <h3>
+      {{RTCDataChannel}} extensions
+    </h3>
+    <p>This document extends {{RTCDataChannel}} by making it <a data-cite="!HTML/#transferable-objects">transferable</a>.</p>
+      This allows sending and receiving messages outside the context the connection was created, for instance in workers or third-party iframes.</p>
+    <div>
+      <p>The WebIDL changes are the following:
+      <pre class="idl"
+>[Exposed=(Window,Worker), Transferable]
+partial interface RTCDataChannel {
+};</pre>
+    </div>
+    <div>
+      <p>The <a data-cite="WEBRTC#dfn-create-an-rtcdatachannel">create an RTCDataChannel</a> algorithm is updated by adding the following steps after step 4 of the original algorithm:</p>
+      <ol>
+        <li><p>Initialize <var>channel</var>.`[[IsTransferable]]` to <code>true</code>.</p></li>
+        <li><p>Queue a task to run the following step:</p>
+          <ol>
+            <li><p>Set <var>channel</var>.`[[IsTransferable]]` to <code>false</code>.</p></li>
+          </ol>
+        </li>
+      </ol>
+    </div>
+    <div>The <a data-cite="WEBRTC#datachannel-send">send() algorithm</a> is modified by adding the following step after step 1 of the original algorithm:
+      <ol>
+        <li><p>Set <var>channel</var>.`[[IsTransferable]]` to <code>false</code>.</p></li>
+      </ol>
+    </div>
+    <div>
+      <p>The {{RTCDataChannel}} <a data-cite="!HTML/#transfer-steps">transfer steps</a>, given <var>value</var> and <var>dataHolder</var>, are:</p>
+      <ol>
+        <li><p>If <var>value</var>.`[[IsTransferable]]` is <code>false</code>, throw a "DataCloneError" DOMException.</p></li>
+        <li><p>Set <var>dataHolder</var>.`[[ReadyState]]` to <var>value</var>.<a href="WEBRTC#dfn-readystate">`[[ReadyState]]`</a>.</p></li>
+        <li><p>Set <var>dataHolder</var>.`[[BufferedAmount]]` to <var>value</var>.<a href="WEBRTC#dfn-bufferedamount">`[[BufferedAmount]]`</a>.</p></li>
```

Comment:
I dont' think we should transfer bufferedAmount, because that value could change.

Reply:
That value should always be 0, otherwise channel is not transferable

Reply:
Then there's no need to transfer it.

Good that you added the prohibition against send() explicitly - I was reviewing the version before that change.

Reply:
Or even simpler: Don't include it in the dataHolder.

Reply:
Done